src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedTagStore.java

```diff
@@ -48,7 +69,7 @@ protected PackedTagStore(final long maximumSize, final int memoryBlockSize,
      */
     public void add(final long index, final String key, final String value)
     {
-        if (index > this.index)
+        if (index > size())
         {
             throw new CoreException(
```

Comment:
Nit: not part of your changes, but consider updating the `CoreException` to follow insertion format:
`throw new CoreException("Cannot add. Invalid index {} is bigger than the size {}", index, size());`